### PR TITLE
Comments v1

### PR DESCRIPTION
**Pull request by jasonify**
Still needed but not part of this PR:
[ ] delete / update
[ ] pagination 
[ ] figure out how we will implement Usernames+Real Names so we can launch to prod
Tests:
[ ] need to check that when no content field is provided it throws back an error

### CODE REVIEW

**Comment by jasonify** [comment]
Hmm, not sure why Travis CI is failing? Will have to look into that.

File: server/tests/comment.test.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        .post(`/api/posts/${postId}/comment`)
+        .set('Authorization', `Bearer ${userToken}`)
+        .send({content})
+        .expect(httpStatus.CREATED)
+        .then((res) => {
+          expect(res.body).to.exist;
+          const comment = res.body.result;
+          expect(comment.post).to.eql(`${postId}`);
+          expect(comment.content).to.eql(`${content}`);
+          expect(comment.author).to.exist; //eslint-disable-line
```
Comment (jasonify) [inline server/tests/comment.test.js]: Maybe the author should match the id of the user we created

**Comment by TheHollidayInn** [comment]
I have fixed Travis. Can you merge in master and push so to see if tests pass?

**Comment by jasonify** [comment]
Seems to work now :)

File: server/models/comment.model.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        if (comment) {
+          return commment;
+        }
+        const err = new APIError('No such comment exists!', httpStatus.NOT_FOUND);
+        return Promise.reject(err);
+      });
+  },
+  getCommentsForItem(postId) {
+    return this.find({post: postId })
+      .sort({dateCreated: -1})
```
Comment (jasonify) [inline server/models/comment.model.js]: Should probably be in reverse order now..

File: server/models/comment.model.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      .exec()
+      .then((comment) => {
+        if (comment) {
+          return commment;
+        }
+        const err = new APIError('No such comment exists!', httpStatus.NOT_FOUND);
+        return Promise.reject(err);
+      });
+  },
+  getCommentsForItem(postId) {
```
Comment (jasonify) [inline server/models/comment.model.js]: Item and postId should be consistent naming.

File: server/models/comment.model.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+/**
+ * Statics
+ */
+CommentSchema.statics = {
+  /**
+   * Get user
+   * @param {ObjectId} id - The objectId of the comment.
+   * @returns {Promise<Comment, APIError>}
+   */
+  get(id) {
```
Comment (jasonify) [inline server/models/comment.model.js]: This method technically doesn't get used. Do we really need it?

**Pull request merged**